src/CommandLineUtils/Internal/ValueParsers/ValueParser.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Nate McMaster.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace McMaster.Extensions.CommandLineUtils.Abstractions
+{
+    using System;
+    using System.Globalization;
+
+    /// <summary>
+    /// Provides methods for creating <see cref="IValueParser{T}"/>
+    /// boilerplate implementations.
+    /// </summary>
+
+    public static class ValueParser
+    {
+        /// <summary>
+        /// Creates an <see cref="IValueParser"/> implementation for a type
+        /// given a parsing function that receives an argument name, a value
+        /// to parse and a culture to use for parsing.
+        /// </summary>
+
+        public static IValueParser Create(Type targetType, Func<string, string, CultureInfo, object> parser) =>
+            new DelegatingValueParser(targetType, Create(parser));
+
+        /// <summary>
+        /// Creates an <see cref="IValueParser{T}"/> implementation given
+        /// a parsing function that receives an argument name, a value to
+        /// parse and a culture to use for parsing.
+        /// </summary>
+
+        public static IValueParser<T> Create<T>(Func<string, string, CultureInfo, T> parser) =>
```

Comment:
The signature is derived from `IValueParser<>`'s `Parse`:

    T Parse(string argName, string value, CultureInfo culture);

I think the purpose of `ValueParse` is very clear and so there should be no surprise about the arguments and their order. The problem with introducing a delegate for the purpose of argument naming is that we'd have to introduce many such delegates, one for each distinct case in the `Create` overloads. In the XML doc comments, I've tried my best to be clear and thorough:

```c#
        /// <summary>
        /// Creates an <see cref="IValueParser{T}"/> implementation given
        /// a parsing function that receives an argument name, a value to
        /// parse, a culture to use for parsing and returns a tuple whose
        /// first element indicates whether parsing was successful and
        /// second element is the parsed value. An additional parameter
        /// specifies a function that returns the <see cref="FormatException"/>
        /// to throw, given the argument name and value, when parsing is
        /// unsuccessful.
        /// </summary>
```

See especially the last bit, “An additional parameter specifies a function that returns the `FormatException` to throw, _given the argument name and value_, when parsing is unsuccessful.”

In hindsight, it might have been better to have `IValueParser<>`'s `Parse` take a context as an argument, which would have hosted the argument name as its property. Then, even a delegate would have been clear, reading as `Func<ValueParseContext, string, CultureInfo, object>`. While it's too late to change this for the interface, it can still be done for `ValueParser.Create` overloads if it helps to make signature clearer through types. Thoughts?

Reply:
Thanks for your thoughtful response. I'm convinced - we can keep this PR as is.

I agree with your hindsight and the context class. For now, I'm okay leaving this as. If we need to add more to value parsing inputs in the future, I'll reconsider this.